Source/WebInspectorUI/UserInterface/Models/TimelineRecording.js

```diff
@@ -85,21 +81,21 @@ WI.TimelineRecording = class TimelineRecording extends WI.Object
         recording._endTime = endTime;
         recording._discontinuities = discontinuities;
 
-        recording.updateCallingContextTrees(target, sampleStackTraces, sampleDurations);
-
-        let topDownCallingContextTree = undefined;
+        if (samples) {
+            for (let {target, stackTraces, durations} of samples)
+                recording.updateCallingContextTrees(WI.ImportedTarget.import(target), stackTraces, durations);
+        } else if (json.sampleStackTraces && json.sampleDurations)
```

Comment:
Add comment about using these keys for compatibility with older saved exported timeline recordings.